### PR TITLE
[2747] bug changing an existing trainees route to other errors

### PR DESCRIPTION
**Pull request by JonF1**
### Context
https://trello.com/c/4meZO55D/2747-bug-changing-an-existing-trainees-route-to-other-errors
Fixes two bugs
1) Error on selecting "other" as a training route on an existing trainee
2) When the user goes to not supported route view after selecting other, the go back link takes the user to the `/trainees/new` page. If the trainee has already been created and the route simply needs to be edited, this is not correct. 

### Changes proposed in this pull request
* The go back link will take user to `trainees/#{trainee_id}/training-routes/edit` if trainee exists. If a new trainee then `/trainees/new` as before
* Update trainee method returned if 'other' selected to avoid error

### Guidance to review
Edit an existing trainees route and select 'other' option. And also create a new trainee and do the same.

### CODE REVIEW

**Comment by JonF1** [comment]
[Bug - changing an existing trainee's route to 'other' errors](https://trello.com/c/4meZO55D/2747-bug-changing-an-existing-trainees-route-to-other-errors)

File: app/controllers/trainees/training_routes_controller.rb [hunk 1]
```diff
@@ -10,12 +10,14 @@ def edit; end
     def update
       trainee.update_training_route!(training_route)
 
-      redirect_url
+      redirect_url(training_route)
```
Comment (defong) [inline app/controllers/trainees/training_routes_controller.rb:13]: You could short-circuit it here

```
if training_route == "other"
  redirect_to trainees_not_supported_route_path(trainee_id: trainee)
else
  redirect_url
end

```

Comment (JonF1) [reply app/controllers/trainees/training_routes_controller.rb]: thanks, fixed

File: app/lib/route_data_manager.rb [hunk 2]
```diff
@@ -6,6 +6,8 @@ def initialize(trainee:)
   end
 
   def update_training_route!(route)
+    return if route == "other"
```
Comment (defong) [inline app/lib/route_data_manager.rb:9]:
```ruby
RouteDataManager.new(trainee: Trainee.first).update_training_route! "other"

# ArgumentError: 'other' is not a valid training_route
```

You don't really need this as the controller has guarded that

Comment (JonF1) [reply app/lib/route_data_manager.rb]: thanks

**Review by jhenderson — APPROVED** [review]

**Comment by bbelward** [comment]
tested and that's working, will card up a separate ticket to change the copy on the page as it currently says we'll be "adding more routes soon" when we won't be! 

this one's good to merge though.

**Pull request merged**